faust/stores/aerospike.py

```diff
@@ -0,0 +1,198 @@
+import typing
+from typing import Any, Dict, Iterator, Optional, Tuple, Union
+
+try:  # pragma: no cover
+    import aerospike
+except ImportError:  # pragma: no cover
+    aerospike = None  # noqa
+
+from yarl import URL
+
+from faust.stores import base
+from faust.types import TP, AppT, CollectionT
+
+if typing.TYPE_CHECKING:  # pragma: no cover
+    from aerospike import SCAN_PRIORITY_MEDIUM, TTL_NEVER_EXPIRE, Client
+else:
+
+    class Client:  # noqa
+        """Dummy Client."""
+
+    TTL_NEVER_EXPIRE = -1
+    SCAN_PRIORITY_MEDIUM = 2
+
+
+if typing.TYPE_CHECKING:  # pragma: no cover
+    import aerospike.exception.RecordNotFound
+else:
+
+    class RecordNotFound:  # noqa
+        """Dummy Client."""
+
+
+aerospike_client: Client = None
+
+
+class AeroSpikeStore(base.SerializedStore):
+    """Aerospike table storage."""
+
+    client: Client
+    BIN_KEY = "value_key"
+    namespace: str = ""
+
+    def __init__(
+        self,
+        url: Union[str, URL],
+        app: AppT,
+        table: CollectionT,
+        options: typing.Mapping[str, Any] = None,
+        **kwargs: Any,
+    ) -> None:
+        try:
+            self.client = AeroSpikeStore.get_aerospike_client(options)
+            self.namespace = options.get("namespace", "")
+        except Exception as ex:
+            self.logger.error(f"Error configuring aerospike client {ex}")
+            raise ex
+        super().__init__(url, app, table, **kwargs)
+
+    @staticmethod
+    def get_aerospike_client(aerospike_config: Dict[Any, Any]) -> Client:
```

Comment:
I think I'm seeing a typing discrepancy between the function parameters for `AeroSpikeStore::__init__(options)` and `AeroSpikeStore::get_aerospike_client(get_aerospike_client)`. The `__init__` defaults to `None` for `options`, but `options` is then used for the `aerospike_config` parameter of `get_aerospike_client` which is _not_ optional.

Reply:
good call